### PR TITLE
Fix deleted product slug handling to be simpler

### PR DESCRIPTION
**Pull request by mbj**

### CODE REVIEW

**Comment by mbj** [comment]
@dkubb Ready for review.

File: core/app/models/spree/product.rb [hunk 1]
```diff
@@ -268,7 +268,7 @@ def normalize_slug
 
     def punch_slug
       # punch slug with date prefix to allow reuse of original
```
Comment (dkubb) [inline core/app/models/spree/product.rb]: This comment no longer matches the functionality of the code.

Comment (mbj) [inline core/app/models/spree/product.rb]: Good catch. I'll adapt it.

Comment (mbj) [inline core/app/models/spree/product.rb]: Ideally we simply change the schema to not assert uniqueness on deleted products. This is out of scope for the current "master iteration" in #127. So I'll skip it for now and instead fix the method name.

**Comment by mbj** [comment]
@dkubb Ready for your next pass.

**Pull request merged**